aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -784,13 +1255,52 @@ module aptos_framework::delegation_pool {
     /// if existing and already inactive to allow the creation of a new one.
     /// `pending_inactive` stake would be left untouched even if withdrawable and should
     /// be explicitly withdrawn by delegator
-    fun execute_pending_withdrawal(pool: &mut DelegationPool, delegator_address: address) {
+    fun execute_pending_withdrawal(pool: &mut DelegationPool, delegator_address: address) acquires GovernanceRecords {
         let (withdrawal_exists, withdrawal_olc) = pending_withdrawal_exists(pool, delegator_address);
         if (withdrawal_exists && withdrawal_olc.index < pool.observed_lockup_cycle.index) {
             withdraw_internal(pool, delegator_address, MAX_U64);
         }
     }
 
+    /// Buy shares into the active pool on behalf of delegator `shareholder` who
+    /// deposited `coins_amount`. This function doesn't make any coin transfer.
+    fun buy_in_active_shares(
+        pool: &mut DelegationPool,
+        shareholder: address,
+        coins_amount: u64,
+    ): u128 acquires GovernanceRecords{
+        let new_shares = pool_u64::amount_to_shares(&pool.active_shares, coins_amount);
+        // No need to buy 0 shares.
+        if (new_shares == 0) { return 0 };
+
+        // Always update governance records before any change to the shares pool.
+        let pool_address = get_pool_address(pool);
+        // <active shares> of <shareholder> += <new_shares> ---->
+        // <active shares> of <current voter of shareholder> += <new_shares>
+        // <active shares> of <next voter of shareholder> += <new_shares>
+        if (partial_governance_voting_enabled(pool_address)) {
```

Comment:
can this logic be enclosed in a separate function, possibly supporting subtraction too (in case of redeeming active shares)? It could be called here for addition and in `redeem_active_shares` with subtraction of voting shares.

Reply:
the same for the buy/redeem of pending-inactive voting shares

Reply:
Yeah make sense